Testscripts/Linux/SetupHbKernel.sh

```diff
@@ -147,17 +153,51 @@ function Main() {
 		dracut -f
 		LogMsg "$?: Run dracut -f"
 	else
-		sed -i -e "s/rootdelay=300/rootdelay=300 resume=$sw_uuid/g" /etc/default/grub.d/50-cloudimg-settings.cfg
-		LogMsg "$?: Updated the 50-cloudimg-settings.cfg with resume=$sw_uuid"
+		_entry=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'rootdelay=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/rootdelay=300/rootdelay=300 resume=$sw_uuid/g" /etc/default/grub.d/50-cloudimg-settings.cfg
+			LogMsg "$?: Updated the 50-cloudimg-settings.cfg with resume=$sw_uuid"
+		else
+			_entry=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep GRUB_CMDLINE_LINUX_DEFAULT)
+			if [ -n "$_entry" ]; then
+				sed -i  '/^GRUB_CMDLINE_LINUX_DEFAULT=/ s/"$/ resume='$sw_uuid'"/'  /etc/default/grub.d/50-cloudimg-settings.cfg
+			else
+				echo GRUB_CMDLINE_LINUX_DEFAULT="console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300 resume=$sw_uuid" >> /etc/default/grub.d/50-cloudimg-settings.cfg
+			fi
+			LogMsg "$?: Added resume=$sw_uuid in 50-cloudimg-settings.cfg file"
+		fi
 
-		sed -i -e "s/GRUB_HIDDEN_TIMEOUT=*.*/GRUB_HIDDEN_TIMEOUT=30/g" /etc/default/grub.d/50-cloudimg-settings.cfg
-		LogMsg "$?: Updated GRUB_HIDDEN_TIMEOUT value with 30"
+		_entry=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'GRUB_HIDDEN_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_HIDDEN_TIMEOUT=*.*/GRUB_HIDDEN_TIMEOUT=30/g" /etc/default/grub.d/50-cloudimg-settings.cfg
+			LogMsg "$?: Updated GRUB_HIDDEN_TIMEOUT value with 30"
+		else
+			echo 'GRUB_HIDDEN_TIMEOUT=30' >> /etc/default/grub.d/50-cloudimg-settings.cfg
+			LogMsg "$?: Added GRUB_HIDDEN_TIMEOUT=30 in 50-cloudimg-settings.cfg file"
+		fi
 
-		sed -i -e "s/GRUB_TIMEOUT=*.*/GRUB_TIMEOUT=30/g" /etc/default/grub.d/50-cloudimg-settings.cfg
-		LogMsg "$?: Updated GRUB_TIMEOUT value with 30"
+		_entry=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'GRUB_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_TIMEOUT=.*/GRUB_TIMEOUT=30/g" /etc/default/grub.d/50-cloudimg-settings.cfg
+			LogMsg "$?: Updated GRUB_TIMEOUT value with 30"
+		else
+			echo 'GRUB_TIMEOUT=30' >> /etc/default/grub.d/50-cloudimg-settings.cfg
+			LogMsg "$?: Added GRUB_TIMEOUT=30 in 50-cloudimg-settings.cfg file"
+		fi
 
 		update-grub2
 		LogMsg "$?: Ran update-grub2"
+
+		_entry1=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'resume=')
+		_entry2=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'GRUB_HIDDEN_TIMEOUT=')
+		_entry3=$(cat /etc/default/grub.d/50-cloudimg-settings.cfg | grep 'GRUB_TIMEOUT=')
```

Comment:
GRUB_TIMEOUT=30

Reply:
Fixed in 27d4b9e05ce5a2135f680e9761682646d9054cc4